### ISSUE TITLE
BUG: Mac Safari - Sidebar animation is different reported by @mdneyazahmad

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11809]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Goto staging https://staging.new.expensify.com/
2. Click on fab
3. Click request money
4. Enter the amount
5. Click next, and observe the animation
6. Select the contact, and observe the animation

## Expected Result:
Animation should be same across all sidebars

## Actual Result:
Animation is not same, the sidebar does not come into view from the end of the screen like other sections

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Web

**Version Number:** 1.2.13-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/195650307-6f05428f-904f-4a0f-be8b-f85072bb6c2d.mp4

https://user-images.githubusercontent.com/43996225/195650316-a5bd57e3-6a5d-4dc3-887f-9eefae88e27e.mov

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @mdneyazahmad
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1665552011863769

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by zanyrenney** [comment]
Hey @kavimuru sorry I am not clear on what the bug is here. Could you please add more detail so that I can try to reproduce. The video is not clear to me what the expected vs actual result is so if you can explain it in text so I can follow the guide to reproduce, that would be really helpful - thank you!

**Comment by kavimuru** [comment]
@zanyrenney I am attaching a new video. The sidebar animation appears from right end of the screen in `profile `page, `preference `page. But when initiating the IOU after entering the money, hitting `Next `no animation and next page appears from left. 

https://user-images.githubusercontent.com/43996225/196058321-c3b00b94-a113-493d-b798-da554e6ced37.mp4

**Comment by zanyrenney** [comment]
Been OOO.

**Comment by thienlnam** [comment]
Going OOO next week, going to re-assign this so we can keep moving forward with proposals

**Comment by Luke9389** [comment]
Looks like we're still waiting on proposals, right?

**Comment by JmillsExpensify** [comment]
I'm not sure if we should fix this. We're going to refactor this entire flow very shortly, so I would be more inclined to put this bug report on hold. At the point we refactor the flow we can see if this is still an issue and then either close (and pay for reporting), or fix. Thoughts guys?

**Comment by mdneyazahmad** [comment]
This is just about consistency. Maybe, the design team can confirm?

**Comment by Luke9389** [comment]
hey @shawnborton,

What do you think about the fact that the animation goes two different directions here? 
Problem, not problem?

**Comment by shawnborton** [comment]
My take is that in the Settings pages, each modal page is animated. But in the IOU flow, just the content of the modal (everything below the title) is being animated. Is that the difference that we're talking about? It's a bit unclear from the comments.

**Comment by JmillsExpensify** [comment]
@shawnborton Great point of clarification. It's basically this.

> The sidebar animation appears from right end of the screen in profile page, preference page. But when initiating the IOU after entering the money, hitting Next, there no animation and next page appears from left.

This is all super subtle.

**Comment by shawnborton** [comment]
Got it. I mean yeah, ideally this would all be consistent and feel silky smooth (IOU animations feel janky on desktop at least).

**Comment by JmillsExpensify** [comment]
100%. Where I struggle is whether we do anything now or re-evaluate once N7 hits. Maybe we just bite the bullet and tackle this now along with all the other bugs. Hmm, yeah that sounds the most logical. Agreed?

**Comment by shawnborton** [comment]
I can get down with that.

**Comment by JmillsExpensify** [comment]
Ok cool. Upwork job is here: https://www.upwork.com/jobs/~010cf9eb9f9b6cb3b2. We're open for proposals.

**Comment by Luke9389** [comment]
waiting for proposals.

**Comment by JmillsExpensify** [comment]
Still waiting for proposals

**Comment by JmillsExpensify** [comment]
Time to raise the price to $500. We're still open for proposals!

**Comment by JmillsExpensify** [comment]
Still waiting on proposals! We're now at $1,000. Upwork job is here: https://www.upwork.com/jobs/~010cf9eb9f9b6cb3b2.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by shawnborton** [comment]
Not overdue

**Comment by JmillsExpensify** [comment]
Going to keep an eye the inability to reproduce, though yeah, for now we are still waiting on proposals.

**Comment by andrewlor** [comment]
# Proposal

## RCA
Each step of the IOU Modal are animated by being wrapped in `AnimatedStep`. The header bar is rendered outside all of these steps, without animation, thus the unnatural look.

## Solution
In `IOUModal`, render the header bar directly adjacent to each page inside of `AnimatedStep`. This animates the header along with the page content for a more natural animation look.

Essentially I'm thinking refactor all of this JSX out into a function or component. For now I just pulled it out into a function `renderHeader`, but could maybe be a component `IOUHeaderBar`?
https://github.com/Expensify/App/blob/7de9ab2d62849fc670879a680dd46cf8003697f8/src/pages/iou/IOUModal.js#L360-L395

Then render it inside of each `AnimatedStep`:
```javascript
<>
    {currentStep === Steps.IOUAmount && (
        <AnimatedStep
            direction={this.getDirection()}
            style={[styles.flex1]}
        >
            {this.renderHeader()}
            <IOUAmountPage
                ...
            />
        </AnimatedStep>
    )}
    {currentStep === Steps.IOUParticipants && (
        <AnimatedStep
            style={[styles.flex1]}
            direction={this.getDirection()}
        >
            {this.renderHeader()}
            <IOUParticipantsPage
                ...
            />
        </AnimatedStep>
    )}
    {currentStep === Steps.IOUConfirm && (
        <AnimatedStep
            style={[styles.flex1]}
            direction={this.getDirection()}
        >
            {this.renderHeader()}
            <IOUConfirmPage
                ...
            />
        </AnimatedStep>
    )}
</>
```

Additionally I had to make a small style adjustment, removed `styles.pageWrapper` from the first `AnimatedStep` for `Steps.IOUAmount` and moved it to `IOUAmountPage` to maintain the padding.

## Result

https://user-images.githubusercontent.com/8475244/202548599-c6fe3c81-144f-48da-8736-a86ba83f2683.mov

**Comment by Luke9389** [comment]
@andrewlor @rushatgabhane 

> The header bar is rendered outside all of these steps, without animation, thus the unnatural look.

This is not the issue we are trying to solve. We are specifically looking at how the animation goes RIGHT -> LEFT sometimes and  LEFT -> RIGHT other times. 

https://github.com/Expensify/App/issues/11809#issuecomment-1280057548

**Comment by andrewlor** [comment]
@Luke9389 Right, but from what I can tell from the given examples, the main content in the modal is always animated in the correct direction. RIGHT -> LEFT if you're going forward a step and LEFT -> RIGHT if you're going back a step, I assumed this behaviour was intentional.

So my interpretation was that it's just the header bar that is animated the wrong direction (or not animated at all). As seen in this example:

https://user-images.githubusercontent.com/8475244/203178222-96491c7a-6ed4-432c-a3bd-f3244a7fcc91.mov

Let me know if that makes sense and is a correct interpretation of the issue.

CC: @rushatgabhane

**Comment by JmillsExpensify** [comment]
Still working through @andrewlor's proposal. I'll circle back tomorrow to see where we land.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by trjExpensify** [comment]
I think @andrewlor is right, the IOUHeader not being in step with the content of the modal is what's making this look strange.

**Comment by mountiny** [comment]
Stepping in to move this forward even ovee Thanksgiving.

Thanks @rushatgabhane for a review 🙌 

Chatted with Tom in DM and i agree the solution @andrewlor proposed achieves what we want here. Can you please raise a PR?

**Comment by andrewlor** [comment]
@rushatgabhane @mountiny PR should be ready to review if you want to take a look, thanks.

**Comment by JmillsExpensify** [comment]
PR is ready for review. @rushatgabhane has made his way through, and we're awaiting @Luke9389's review once he returns from the Thanksgiving holiday in the US.

**Comment by JmillsExpensify** [comment]
Ok the linked PR is merged, and met the criteria of merging within 3 business days in my opinion. Accordingly the payout for this issue is as follows:

 - Contributor: @andrewlor $2,000 (incl. 50% bonus)
 - Contributor+: @rushatgabhane $2,000 (incl. 50% bonus)
 - Issue reporter: @mdneyazahmad $250

Please apply to the [Upwork job](https://www.upwork.com/jobs/~010cf9eb9f9b6cb3b2) so that we're ready to issue payments once the regression period has passed.

**Comment by mdneyazahmad** [comment]
@JmillsExpensify applied. Thanks!

**Comment by JmillsExpensify** [comment]
All contributors hired for this job. Still waiting on the regression period. In the meantime, @rushatgabhane and @Luke9389 can you confirm we're good on those three items in the checklist above? We should be good but let's check that box. Then I need to circle back and create the regression tests in TestRail as well.

**Comment by JmillsExpensify** [comment]
Everyone is paid out at this time. I won't be able to close the loop on regression today so updating the hold accordingly.

**Comment by JmillsExpensify** [comment]
All contributors have been paid out. We just lack the regression test at this point.

**Comment by JmillsExpensify** [comment]
Interesting. Looks like our Upwork integration is preventing me from renaming this issue. That's fun.

**Comment by JmillsExpensify** [comment]
Alright, I created and linked the regression test issue. Checklist above is set, so I'm closing out this issue.
